pkg/agent/cniserver/pod_configuration_windows.go

```diff
@@ -0,0 +1,369 @@
+// Copyright 2020 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build windows
+
+package cniserver
+
+import (
+	"fmt"
+	"net"
+	"strings"
+	"sync"
+
+	"github.com/Microsoft/hcsshim"
+	cnitypes "github.com/containernetworking/cni/pkg/types"
+	"github.com/containernetworking/cni/pkg/types/current"
+	"k8s.io/klog"
+
+	"github.com/vmware-tanzu/antrea/pkg/agent/interfacestore"
+	"github.com/vmware-tanzu/antrea/pkg/agent/openflow"
+	"github.com/vmware-tanzu/antrea/pkg/agent/util"
+	cnipb "github.com/vmware-tanzu/antrea/pkg/apis/cni/v1beta1"
+	"github.com/vmware-tanzu/antrea/pkg/ovs/ovsconfig"
+)
+
+const (
+	notFoundHNSEndpoint = "The endpoint was not found"
+)
+
+type endpoint struct {
+	hnsEP      *hcsshim.HNSEndpoint
+	containers []string
+}
+
+type podConfigurator struct {
+	ovsBridgeClient ovsconfig.OVSBridgeClient
+	ofClient        openflow.Client
+	ifaceStore      interfacestore.InterfaceStore
+	gatewayMAC      net.HardwareAddr
+	ovsDatapathType string
+	hnsNetwork      *hcsshim.HNSNetwork
+	epCache         *sync.Map
+}
+
+func newPodConfigurator(
+	ovsBridgeClient ovsconfig.OVSBridgeClient,
+	ofClient openflow.Client,
+	ifaceStore interfacestore.InterfaceStore,
+	gatewayMAC net.HardwareAddr,
+	ovsDatapathType string,
+) *podConfigurator {
+	epCache := &sync.Map{}
+	eps, err := hcsshim.HNSListEndpointRequest()
+	if err != nil {
+		klog.Fatalf("Failed to list existing HNS Endpoints, using an empty cache: %v", err)
```

Comment:
Fatalf will terminate the process, is it expected? even if it's expected, "using an empty cache" doesn't make sense. I would suggest make the function returning error and exiting from cmd consistently.

Reply:
I didn't mean to change the log level to "fatal" by "being fatal". I just meant the error is critical to this process and it cannot continue so should exit, but I prefer letting its consumer do it to have a unified workflow and prevent this case from being not unit testable as the function has the side effect of exiting the process. Sorry for being confusing.